_data-prepper/expression-syntax.md

```diff
@@ -0,0 +1,193 @@
+| 1     | `and`, `or`          | Conditional Expression                                | left-to-right |
+
+# Reserved for possible future functionality
+Reserved symbol set: `^`, `*`, `/`, `%`, `+`, `-`, `xor`, `=`, `+=`, `-=`, `*=`, `/=`, `%=`, `++`, `--`, `${<text>}`
+
+# Set initializer
+Defines a set or term and/or expressions.
+
+## Examples
+
+### HTTP status codes
+```
+{200, 201, 202}
+```
+### HTTP response payloads
+```
+{"Created", "Accepted"}
+```
+
+### Handle multiple event types with different keys
+```
+{/request_payload, /request_message}
+```
+
+# Priority expression
+Identifies an expression that will be evaluated at the highest priority level. Priority expression must contain an
+expression or value, empty parentheses are not supported.
+
+## Examples
+
+```
+/is_cool == (/name == "Steven")
+```
+
+# Relational operators
+Tests the relationship of two numeric values. Note, the operands must be a number or Json Pointer that will resolve to a number.
+
+## Syntax
+```
+<Number | Json Pointer> < <Number | Json Pointer>
+<Number | Json Pointer> <= <Number | Json Pointer>
+<Number | Json Pointer> > <Number | Json Pointer>
+<Number | Json Pointer> >= <Number | Json Pointer>
+```
+
+## Examples
+```
+/status_code >= 200 and /status_code < 300
+```
+
+# Equality operators
+Used to test if two values are equivalent or not equivalent.
+
+## Syntax
+```
+<Any> == <Any>
+<Any> != <Any>
+```
+
+## Examples
+```
+/is_cool == true
+3.14 != /status_code
+{1, 2} == /event/set_property
+```
+# Using equality operators to check JSON pointer 
+
+Equality operators can also be used to check whether a JSON Pointer exists by checking with 'null'
+
+## Syntax
+```
+<Json Pointer> == null
+<Json Pointer> != null
+null == <Json Pointer>
+null != <Json Pointer>
+```
+
+## Examples
+```
+/response == null
+null != /response
+```
+
+# Conditional expression
+Used to chain together multiple expressions and/or values.
+
+## Syntax
+```
+<Any> and <Any>
+<Any> or <Any>
+not <Any>
+```
+
+## Examples
+```
+/status_code == 200 and /message == "Hello world"
+/status_code == 200 or /status_code == 202
+not /status_code in {200, 202}
+/response == null
+/response != null
+```
+
+# Definitions
+## Literal
+A fundamental value that has no children.
+- Float _(Supports values from 3.40282347 x 10^38 to 1.40239846 x 10^-45)_
+- Integer _(Supports values from -2147483648 to 2147483647)_
+- Boolean _(Supports true or false)_
+- Json Pointer _(See Json Pointer section for details)_
+- String _(Supports Valid Java String characters)_
+- Null _(Supports null check to see if a Json Pointer is present or not)_
+
+## Expression string
+The String that will be parsed for evaluation. Expression String is the highest level of a Data Prepper Expression. Only supports one expression string resulting in a return value. Note, an _Expression String_ is not the same as an _Expression_.
```

Comment:
Should "string" be lowercase here in all instances?

Reply:
@kolchfa-aws Yes, I think it should.